client/src/components/SendMessage.js

```diff
@@ -36,36 +39,73 @@ class SendMessage extends React.Component {
     return;
   }
 
+  hideEmojiPicker() {
+      this.setState({
+          ...this.state,
+          emojiPickerActive : false
+      });
+  }
+  onKeyUp(event) {
+      if (this.state.emojiPickerActive){
+          if (event.which === 9 || event.which === 37 || event.which === 39) {
+              // Cycle through preview
+              this.refs.emojipicker.onKeyUp(event);
+          } else {
+              const lastWord = this.refs.message.value.split(" ").pop();
+              this.refs.emojipicker.filterWord(lastWord);
+          }
+      }
+  }
+
   onKeyDown(event) {
```

Comment:
With all the refactoring from the feedback, we can simplify the onKeyDown down to:
```javascript
  onKeyDown(event) {
    if(this.state.emojiPickerActive)
      this.refs.emojipicker.onKeyDown(event);
  }
```